BoxContentSDK/BoxContentSDK/BOXURLSessionCacheClient.m

```diff
@@ -0,0 +1,297 @@
+    return [self cacheBackgroundSessionId:backgroundSessionId sessionTaskId:sessionTaskId data:data type:BOXURLSessionTaskCacheFileTypeError error:error];
+}
+
+- (BOXURLSessionTaskCachedInfo *)getCacheForUserId:(NSString *)userId associateId:(NSString *)associateId error:(NSError **)error
+{
+    BOXURLBackgroundSessionIdAndSessionTaskId *backgroundSessionIdAndSessionTaskId = [self backgroundSessionIdAndSessionTaskIdGivenUserId:userId associateId:associateId];
+
+    if (backgroundSessionIdAndSessionTaskId == nil) {
+        if (error != nil) {
+        (*error) = [[NSError alloc] initWithDomain:BOXURLSessionTaskCacheError code:BOXContentSDKSCacheErrorNoValidSessionTaskGivenUserIdAssociateId userInfo:nil];
+        }
+        return nil;
+    }
+
+    NSString *backgroundSessionId = backgroundSessionIdAndSessionTaskId.backgroundSessionId;
+    NSUInteger sessionTaskId = backgroundSessionIdAndSessionTaskId.sessionTaskId;
+
+    NSString *dir = [self dirPathOfSessionTaskWithBackgroundSessionId:backgroundSessionId sessionTaskId:sessionTaskId];
+
+    BOXURLSessionTaskCachedInfo *cachedInfo = [[BOXURLSessionTaskCachedInfo alloc] init];
+
+    //get all files under sessions/$backgroundSessionId/$sessionTaskId
+    NSArray *files = [[NSFileManager defaultManager] contentsOfDirectoryAtPath:dir error:error];
+    for (int i = 0; i < files.count; i++) {
+        NSString *fileName = files[i];
+        NSString *filePath = [dir stringByAppendingPathComponent:fileName];
+
+        //decrypt data found at filePath
+        NSData *data = [NSData dataWithContentsOfFile:filePath];
+        if ([self.delegate respondsToSelector:@selector(decryptData:)]) {
+            data = [self.delegate decryptData:data];
+        }
+
+        //parse decrypted data based on its file name
+        if ([fileName isEqualToString:BOXURLSessionTaskCacheDestinationFilePath]) {
+            cachedInfo.destinationFilePath = [NSKeyedUnarchiver unarchiveObjectWithData:data];
+        } else if ([fileName isEqualToString:BOXURLSessionTaskCacheResumeData]) {
+            cachedInfo.resumeData = data;
+        } else if ([fileName isEqualToString:BOXURLSessionTaskCacheResponse]) {
+            cachedInfo.response = [NSKeyedUnarchiver unarchiveObjectWithData:data];
+        } else if ([fileName isEqualToString:BOXURLSessionTaskCacheResponseData]) {
+            cachedInfo.responseData = data;
+        } else if ([fileName isEqualToString:BOXURLSessionTaskCacheError]) {
+            cachedInfo.error = [NSKeyedUnarchiver unarchiveObjectWithData:data];
+        }
+    }
+    return cachedInfo;
+}
+
+- (BOOL)deleteCachedInfoForUserId:(NSString *)userId associateId:(NSString *)associateId error:(NSError **)error
+{
+    BOXURLBackgroundSessionIdAndSessionTaskId *backgroundSessionIdAndSessionTaskId = [self backgroundSessionIdAndSessionTaskIdGivenUserId:userId associateId:associateId];
+    if (backgroundSessionIdAndSessionTaskId == nil) {
+        return YES;
```

Comment:
added the check for error from backgroundSessionIdAndSessionTaskIdGivenUserId method, if there is an error, we return NO instead